packages/kbn-scout-reporting/src/reporting/report/events/persistence/index.ts

```diff
@@ -141,3 +145,34 @@ export class ScoutReportDataStream {
     }
   }
 }
+
+/**
+ * Upload events logged by a Scout reporter to the configured Scout Reporter ES instance
+ *
+ * @param eventLogPath Path to event log file
+ * @param log Logger instance
+ */
+export async function uploadScoutReportEvents(eventLogPath: string, log?: ToolingLog) {
+  const logger = log || new ToolingLog();
+
+  const warnSettingWasNotConfigured = (settingName: string) =>
+    logger.warning(`Won't upload Scout reporter events: ${settingName} was not configured`);
+
+  if (SCOUT_REPORTER_ES_URL === undefined) {
+    warnSettingWasNotConfigured('SCOUT_REPORTER_ES_URL');
+    return;
+  }
+
+  if (SCOUT_REPORTER_ES_API_KEY === undefined) {
+    warnSettingWasNotConfigured('SCOUT_REPORTER_ES_API_KEY');
+    return;
+  }
+
+  const es = new ESClient({
+    node: SCOUT_REPORTER_ES_URL,
+    auth: { apiKey: SCOUT_REPORTER_ES_API_KEY },
+  });
```

Comment:
In the CLI `packages/kbn-scout-reporting/src/cli/upload_events.ts`, we're connecting to ES like this:
```
    const es = await getValidatedESClient(
      {
        node: esURL,
        auth: { apiKey: esAPIKey },
        tls: {
          rejectUnauthorized: flagsReader.boolean('verifyTLSCerts'),
        },
      },
      log
    );
```
I'm wondering if we a/ also want to validate the ES connection here and b/ if we also want to pass the TLS options. What do you think?

Reply:
Resolved in https://github.com/elastic/kibana/pull/206431/commits/b95eeeef5d52e508ebf20f5bc1e314ada2260bbd